app/services/projects/download_service.rb

```diff
@@ -0,0 +1,40 @@
+module Projects
+  class DownloadService < BaseService
+
+    WHITELIST = %w(fogbugz.com)
+
+    def initialize(project, url)
+      @project, @url = project, url
+    end
+
+    def execute
+      return nil unless valid_url?(@url)
+
+      uploader = FileUploader.new(@project)
+      uploader.download!(@url)
+      uploader.store!
+
+      filename = uploader.image? ? uploader.file.basename : uploader.file.filename
+
+      {
+        'alt'       => filename,
+        'url'       => uploader.secure_url,
+        'is_image'  => uploader.image?
+      }
+    end
+
+    private
+
+    def valid_url?(url)
+      url && http?(url) && valid_domain?(url)
+    end
+
+    def http?(url)
+      url =~ /\A#{URI::regexp(['http', 'https'])}\z/
+    end
+
+    def valid_domain?(url)
+      WHITELIST.include?(Domainator.parse(url))
```

Comment:
I'm not sure of an easy way to handle subdomains and TLDs such as `co.uk`

Perhaps do the whitelist as regex?

In the case of fogbugz the whitelist needs to allow `*.fogbugz.com`, every account has its own subdomain.